app/src/lumen/java/com/cliqz/browser/purchases/trial/TrialPeriodLocalRepo.kt

```diff
@@ -1,21 +1,34 @@
 package com.cliqz.browser.purchases.trial
 
 import acr.browser.lightning.preference.PreferenceManager
-import com.google.gson.Gson
-import com.google.gson.GsonBuilder
+import android.util.Base64
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.ObjectInputStream
+import java.io.ObjectOutputStream
 
 class TrialPeriodLocalRepo(private val preferenceManager: PreferenceManager) {
 
-    // TODO: Use Base64 encoding or Android X's EncryptedSharedPreferences
-    private val gson: Gson = GsonBuilder().serializeNulls().create()
-
     fun loadPurchaseInfo(trialPeriodResponseListener: TrialPeriodResponseListener) {
-        val trialPeriod = gson.fromJson<TrialPeriod>(
-                preferenceManager.trialPeriodInfo, TrialPeriod::class.java)
-        trialPeriodResponseListener.onTrialPeriodResponse(trialPeriod)
+        if (preferenceManager.trialPeriodInfo == null) {
+            return
```

Comment:
`trialPeriodResponseListener.onTrialPeriodResponse(null)` before this, because the passed listener needs to be invoked (it makes a network call).
Otherwise, I checked, it works as expected